message SystemConfigurationUplinkPayload {
   required bool eps_board_available = 1 [default = true];
   required bool cdh_board_available = 2 [default = true];
   required bool fs_board_available = 3 [default = true];
   required bool tcom_board_available = 4 [default = true];
   required bool utilities_board_available = 5 [default = true];
   required bool x_pos_available = 6 [default = true];
   required bool x_neg_available = 7 [default = true];
   required bool y_pos_available = 8 [default = true];
   required bool y_neg_available = 9 [default = true];
   required bool z_pos_available = 10 [default = true];
   required bool z_neg_available = 11 [default = true];
   required bool i2c_available = 12 [default = true];
   required bool sd_card_available = 13 [default = true];
   required bool bms_available = 14 [default = true];
   required bool antenna_available = 15 [default = false];
   required bool eeprom_available = 16 [default = false];
   required bool enter_deep_sleep_on_startup = 17 [default = false];
   required bool enter_limp_mode_on_startup = 18 [default = false];
   required bool deploy_antenna = 19 [default = false];  
   required bool instant_deployment_waits = 20 [default = true];
   required bool format_sd_on_startup = 21 [default = false];
   required fixed32 beacon_wait_ms = 22 [default = 300000];
   required fixed32 ejection_wait_ms = 23 [default = 300000];
   required fixed64 time_source_deploy_ms = 24 [default = 0];
   required bool lithium_transmit_only_when_ground_commanded = 25 [default = false];
   required fixed32 nominal_lithium_power_level = 26 [default = 140];  
   required bool uplink_fec_enabled = 27 [default = true];
   required bool downlink_fec_enabled = 28 [default = false];
   required bool check_hmac_default = 29 [default = true];
   required bool check_sequence_default = 30 [default = false];
   required bool log_i2c = 31 [default = false];
   required bool log_to_uart = 32 [default = true];
   required bool log_to_sd = 33 [default = true];
   required bool hil_available = 34 [default = false];
   required bool ditl_mode = 35 [default = false];
   required bool verbose_logging = 36 [default = false];
   required bool reboot_on_top_level_exception = 37 [default = true];
   required fixed32 nominal_beacon_period_ms = 38 [default = 10000];
   required fixed32 orientation_control_loop_period_micros = 39 [default = 1000000];
   required fixed32 health_check_period_ms = 40 [default = 1100];
   required fixed64 antenna_burn_check_interval_ms = 41 [default = 60000];
   required fixed32 time_update_period_ms = 42 [default = 1000];
   required fixed32 i_charge_hold_time_ms = 43 [default = 5000];
   required fixed32 i_charge_hold_poll_ms = 44 [default = 200];
   required fixed32 solar_power_recovery_time_ms = 45 [default = 300];
   required fixed32 initial_i_charge_index = 46 [default = 1];
   required fixed64 half_orbit_period_ms = 47 [default = 2790000];
   required fixed64 initial_antenna_burn_interval_ms = 48 [default = 4790000];
   required fixed64 backup_antenna_burn_interval_ms = 49 [default = 8370000];
   required float antenna_burn_interval_multiplier = 50 [default = 2.5];
   required bool run_magnetorquers_at_constant_power = 51 [default = true];
   required float magnetorquer_power_fraction_x = 52 [default = 0.2];
   required float magnetorquer_power_fraction_y = 53 [default = 0.2];
   required float magnetorquer_power_fraction_z = 54 [default = 0.2];
   required double max_magnetorquer_dipole_1 = 55 [default = 0.2];
   required double max_magnetorquer_dipole_2 = 56 [default = 0.2];
   required double max_magnetorquer_dipole_3 = 57 [default = 0.2];
   required double orientation_control_power_level = 58 [default = 0.2];
   required fixed32 checksum = 59 [default = 0];
}
